File: day01/Day01.py
from sys import argv, stdin
import operator as op

xs = [int(l) for l in stdin.readlines()]

solve1 = lambda xs: sum(map(op.lt, xs, xs[1:]))

# Part two pre-proc
if argv[1] == "2":
    xs = list(map(lambda *l: sum(l), xs, xs[1:], xs[2:]))

print(solve1(xs))

File: day01/Day01.rs
#![feature(array_windows)]
use std::env;
use std::io;

fn main() {
    let mut nums = io::stdin()
        .lines()
        .map(|s| s.unwrap().parse::<u32>().unwrap())
        .collect::<Vec<_>>(); // 1000 elements won't hurt

    if env::args().nth(1).unwrap() == "2" {
        // Part two
        nums = nums
            .array_windows::<3>()
            .map(|&w| w.iter().sum())
            .collect::<Vec<_>>();
    }

    let result = nums
        .array_windows::<2>() //
        .filter(|&w| w[0] < w[1])
        .count();

    println!("{}", result);
}
